.github/workflows/lint.yml

```diff
@@ -0,0 +1,38 @@
+name: Lint 
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+permissions:
+  contents: read
+
+concurrency:
+  group: ci-${{ github.ref }}-linting
+  cancel-in-progress: true
+
+jobs:
+  rust:
```

Comment:
Do we want to run golangci-lint for the go part?

Reply:
yes, wanted to add it in a different PR